### ISSUE TITLE
Produce training pairs for siamese nn

### ISSUE THREAD

**Issue by AmirAlavi** [AmirAlavi/single_cell_reducer#3]
Given the original training data of (expression_vector, cell_type), create a new training set of:
( (expression_vector1, expression_vector2), same/different)

This will be the training data for the siamese neural networks. Be sure to have an equal number of same/different labeled samples.

**Comment by AmirAlavi** [comment]
- [ ] Create code to convert a feature matrix X and label vector y --> list of pair samples with corresponding labels, such that there is an equal number of same and different classes.
- [ ] Create a function that is aware of which dataset a sample comes from, and only creates training pairs that have samples from different datasets

**Comment by AmirAlavi** [comment]
Implemented in dd83dd2f2d19556198eb9bdf222c59be3b373772
